zinc/src/main/scala/sbt/internal/inc/MixedAnalyzingCompiler.scala

```diff
@@ -100,8 +100,14 @@ final class MixedAnalyzingCompiler(
           val cp: Seq[VirtualFile] = (extraClasspath map { x =>
             config.converter.toVirtualFile(x.toAbsolutePath)
           }) ++ absClasspath
+          val scalacOpts = config.currentSetup.options.scalacOptions
+          val pickleJava = {
+            if (isPickleJava && !scalacOpts.contains("-Ypickle-java"))
+              List("-Ypickle-java")
+            else Nil
+          }
```

Comment:
I don't know if adding -Y flag forcefully would still make it a good default..

Reply:
Hmm...  Yeah.  Just drop the assert, then?  WDYT?

Reply:
This would mess up pipelining. I think the solution is to actually solve the mixed Scala/Java compilation, and let the build tools configure the flags.

Reply:
Why and how would it mess it up?  It's an assertion, so removing it can't actively mess anything up.

Reply:
I don't disagree with eventually finding a solution for pipelining in mixed projects, but we can drop as it breaks your example.

Reply:
When a build tool (currently scripted) correctly hooks up the pipelined tasks, not passing `-Ypickle-java` would create mysterious error saying the symbols (defined in upstream Java subproject) are not found.

Reply:
Yeah, but it's not an outlier: there's about a billion ways you can set up pipelining wrong.